packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenLock.sol

```diff
@@ -0,0 +1,492 @@
+    string private constant ERROR_ZERO_ADDRESS = "Zero address";
+    string private constant ERROR_NOT_AIRNODE = "Not airnode";
+    string private constant ERROR_ZERO_AMOUNT = "Zero amount";
+    string private constant ERROR_INSUFFICIENT_AMOUNT = "Insufficient amount";
+    string private constant ERROR_ALREADY_LOCKED = "Already locked";
+    string private constant ERROR_NOT_LOCKED = "No amount locked";
+    string private constant ERROR_LOCK_PERIOD_NOT_EXPIRED =
+        "Locking period not expired";
+    string private constant ERROR_REQUESTER_BLOCKED = "Requester blocked";
+    string private constant ERROR_REQUESTER_NOT_BLOCKED =
+        "Requester not blocked";
+    string private constant ERROR_AIRNODE_NOT_OPTED_IN = "Airnode not opted in";
+    string private constant ERROR_NOT_ORACLE = "Not oracle";
+    string private constant ERROR_NOT_ORACLE_ADDRESS_SETTER =
+        "Not oracle address setter";
+    string private constant ERROR_NOT_COEFFICIENT_AND_REGISTRY_SETTER =
+        "Not coefficient and registry setter";
+    string private constant ERROR_NOT_OPT_STATUS_SETTER =
+        "Not opt status setter";
+    string private constant ERROR_NOT_BLOCK_WITHDRAW_DESTINATION_SETTER =
+        "Not block withdraw destination setter";
+    string private constant ERROR_NOT_BLOCK_REQUESTER = "Not block requester";
+    string private constant ERROR_NOT_REQUESTER_AUTHORIZER_WITH_MANAGER_SETTER =
+        "Not RequesterAuthorizerWithManagerSetter";
+
+    /// @notice Address of Api3Token
+    address public immutable api3Token;
+
+    /// @notice Address of AirnodeFeeRegistry
+    address public airnodeFeeRegistry;
+
+    /// @notice The price of API3 in terms of USD
+    /// @dev The price has 6 decimal places
+    uint256 public api3PriceInUsd;
+
+    /// @notice A coefficient used to calculate the amount of tokens to be locked
+    /// @dev The multiplier coefficient has 18 decimal places
+    uint256 public multiplierCoefficient;
+
+    /// @notice Represents the locked amounts and the total whitelist counts of Lockers
+    /// for a chainId-airnode-endpointId-requester pair
+    struct TokenLocks {
+        mapping(address => uint256) lockerToLockAmount;
+        uint256 whitelistCount;
+    }
+
+    /// @notice Stores information about all token locks for a
+    /// chainId-airnode-endpointId-requester pair
+    mapping(uint256 => mapping(address => mapping(bytes32 => mapping(address => TokenLocks))))
+        public tokenLocks;
+
+    /// @notice Stores information for blocked airnode-requester pair
+    ///      Stores true if blocked, false otherwise
+    mapping(address => mapping(address => bool))
+        public airnodeToRequesterToBlockStatus;
+
+    /// @notice The Address to which blocked tokens will be withdrawn to
+    address public blockWithdrawDestination;
+
+    /// @notice mapping used to store all the oracle addresses
+    mapping(address => bool) public isOracle;
```

Comment:
This should be implemented as a role adminned by `oracleAddressSetterRole`

Reply:
isn't that what's happening [here ](https://github.com/api3dao/airnode/blob/2584aa6fb3fb2ec419801e5d68fafb309952b39a/packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenLock.sol#L158-L166)?